OpenRA.Mods.Common/Widgets/Logic/Installation/DownloadPackageLogic.cs

```diff
@@ -217,26 +218,36 @@ void OnDownloadProgress(long total, long read, int progressPercentage)
 						try
 						{
 							using (var stream = File.OpenRead(file))
-							using (var z = new ZipFile(stream))
 							{
-								foreach (var kv in download.Extract)
+								var packageLoaderName = $"{download.Type}Loader";
+
+								var packageLoaderType = AppDomain.CurrentDomain
```

Comment:
`var packageLoader = modData.ObjectCreator.CreateObject<IPackageLoader>($"{download.Type}Loader");` should be sufficient